src/web_interface/components/analysis_routes.py

```diff
@@ -121,13 +122,20 @@ def get_update_analysis(self, uid, re_do=False, error=None):
             vendor_list = self.db.frontend.get_vendor_list()
             device_name_dict = self.db.frontend.get_device_name_dict()
 
-        device_class_list.remove(old_firmware.device_class)
-        vendor_list.remove(old_firmware.vendor)
-        device_name_dict[old_firmware.device_class][old_firmware.vendor].remove(old_firmware.device_name)
+        with ConnectTo(self.intercom, self._config) as intercom:
+            plugin_dict = intercom.get_available_analysis_plugins()
+
+        # Add a temporary preset that holds the plugins of the current analysis
+        current_analysis_preset = old_firmware.uid
+        analysis_presets = [current_analysis_preset] + list(self._config['default-plugins'])
 
         previously_processed_plugins = list(old_firmware.processed_analysis.keys())
-        with ConnectTo(self.intercom, self._config) as intercom:
-            plugin_dict = self._overwrite_default_plugins(intercom.get_available_analysis_plugins(), previously_processed_plugins)
+        # FIXME: why is this even in there?
```

Comment:
This snippet replaces the default plugins (default set when uploading a new firmware) with the plugins run previously on the firmware. I'd rather keep it in a function as before. Maybe rename to sth. like `set_selected_plugins_to_previous`.

Reply:
I extracted the logic in a function.